src/sentry/workflow_engine/migration_helpers/issue_alert_dual_write.py

```diff
@@ -202,10 +202,26 @@ def delete_migrated_issue_alert(rule: Rule) -> int | None:
     workflow: Workflow = alert_rule_workflow.workflow
     workflow_id = workflow.id
 
-    delete_workflow(workflow)
-    alert_rule_workflow.delete()
+    try:
+        delete_workflow(workflow)
+    except DataConditionGroup.DoesNotExist:
+        logger.exception(
+            "Deleting workflow but DCG does not exist",
+            extra={"workflow_id": workflow_id, "rule_id": rule.id},
+        )
 
-    return workflow_id
+    try:
+        workflow.refresh_from_db()
+    except Workflow.DoesNotExist:
+        # expected. only delete the lookup if workflow is deleted
+        alert_rule_workflow.delete()
```

Comment:
What's the reason for not deleting the lookup inside of `delete_workflow`?

Reply:
delete_workflow is a standalone function that only deletes the workflow

Reply:
It deletes actions and dataconditions too, I'm not opposed to keeping it how it is btw, just curious since we're deleting other related stuff in there

Reply:
i would expect it to be used when dual write is over, so didn't want to mix `AlertRuleWorkflow` in there